### PR TITLE
Create report model (first approach)

### PR DESCRIPTION
**Pull request by otoro90**

### CODE REVIEW

**Comment by danialeresbar** [comment]
Please remove the Place model and create the migration again. Make the respective changes in the files that have references to this model.
Thanks

**Comment by otoro90** [comment]
Ok, I will do it

File: reports/admin.py [hunk 1]
```diff
@@ -1,3 +1,22 @@
-from django.contrib import admin
+from django.contrib.gis import admin
```
Comment (danialeresbar) [inline reports/admin.py:1]: Please `django.contrib import admin` instead
Thanks

File: reports/admin.py [hunk 2]
```diff
@@ -1,3 +1,22 @@
...
+from django.contrib.admin import register
 
 # Register your models here.
+from harvests.models import Bunch, CategoryBunch, Harvester
```
Comment (danialeresbar) [inline reports/admin.py:5]: You don't need those models. Import only what you need
Thanks

File: reports/models.py [hunk 3]
```diff
@@ -1,3 +1,30 @@
...
+
+    """
+
+    date = models.DateTimeField(auto_now=False, auto_now_add=False)
+    block = models.CharField(max_length = 16, blank=True)
+    lot = models.CharField(max_length = 16, blank=True)
+    parcel = models.ForeignKey(Parcel, on_delete=models.CASCADE)
+    vehicle = models.ForeignKey(Vehicle, on_delete=models.CASCADE)
+    harvesters = models.ManyToManyField(Harvester)
+    bunches = models.ManyToManyField(Bunch)
```
Comment (danialeresbar) [inline reports/models.py:19]: You should use a foreign key to the BunchBatch model to get all these references.
Please update it. Erase your migration and you can generate it again

File: reports/models.py [hunk 4]
```diff
@@ -1,3 +1,30 @@
...
+
+    class Meta:
+        verbose_name = 'Report'
+        verbose_name_plural = 'Reports'
+
+    def __str__(self):
+        if self.name:
+            return self.name
```
Comment (danialeresbar) [inline reports/models.py:27]: This attribute does not exist!
Please use a human-readable attribute for this method
Thanks

File: reports/models.py [hunk 5]
```diff
@@ -1,3 +1,30 @@
...
+        else:
+            return f'Report-{self.parcel}-{self.block}-{self.lot}'
```
Comment (danialeresbar) [inline reports/models.py:29]: Please update this according to the new model attributes
Thanks

**Review by danialeresbar — COMMENTED** [review]
Please check my suggestions!
Thanks

**Comment by otoro90** [comment]
hello, resolved observations

File: reports/admin.py [hunk 6]
```diff
@@ -1,3 +1,21 @@
...
+    fieldsets = (
+        (None, {
+            'fields': (BaseModelAdmin.readonly_fields,)
+        }),
+        ('Report', {
+            'fields': ('date', 'bunchBatch', 'block', 'lot')
+        })
+    )
+    list_display = ('id', 'bunchBatch', 'block', 'lot', 'creation_date', 'update_date')
+    list_filter = ('creation_date', 'is_active')
```
Comment (danialeresbar) [inline reports/admin.py:21]: Please add one blank space in the EOF

**Review by danialeresbar — COMMENTED** [review]
Please check my suggestions!

**Comment by danialeresbar** [comment]
Once this is done we can proceed to integrate

**Comment by otoro90** [comment]
Hello, 
thanks for the observation, this is already solved

**Review by danialeresbar — APPROVED** [review]
Nice!

There are some changes for the model later on, but it's fine for a first approach 😎

**Pull request merged**